pkg/migration/generate.go

```diff
@@ -936,6 +936,14 @@ func (h *Handler) addMonitoringSpec(cluster *corev1.StorageCluster, ds *appsv1.D
 	}
 	cluster.Status.DesiredImages.Telemetry = telemetryImage
 
+	// Telemetry components are removed from PX DaemonSet 2.12+ so it's not a valid cluster setup to migrate.
+	// CCM upgrade to a newer version during migration is not supported, so we disable telemetry if end up here.
+	if pxutil.GetPortworxVersion(cluster).GreaterThanOrEqual(pxutil.MinimumPxVersionCCMGO) {
+		cluster.Spec.Monitoring.Telemetry.Enabled = false
```

Comment:
Let's still enable it if it was enabled.

Reply:
if it's tested working, we should still enable it, otherwise it's ok to disable it.